osu.Game.Rulesets.Osu/Difficulty/OsuPerformanceCalculator.cs

```diff
@@ -110,10 +119,19 @@ protected override PerformanceAttributes CreatePerformanceAttributes(ScoreInfo s
                 effectiveMissCount = Math.Min(effectiveMissCount + countOk * okMultiplier + countMeh * mehMultiplier, totalHits);
             }
 
+            double clockRate = getClockRate(score);
+
+            hitWindowGreat = 80 - 6 * osuAttributes.OverallDifficulty;
```

Comment:
Pretty sure we can get those from difficulty attributes

Reply:
Does this mean that I should add new difficulty attribute for each judgement hitwindow?

Reply:
https://github.com/ppy/osu/pull/30907/commits/4539246d71afa3c0d0ed14537764a98229dd2bfc